src/Dynatrace.OpenTelemetry.Exporter.Metrics/DynatraceMetricsExporter.cs

```diff
@@ -17,75 +17,78 @@
 using System;
 using System.Collections.Generic;
 using System.Diagnostics;
-using System.IO;
 using System.Net.Http;
 using System.Net.Http.Headers;
 using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Logging.Abstractions;
-using Dynatrace.OpenTelemetry.Exporter.Metrics;
 using OpenTelemetry.Metrics.Export;
 
-namespace Dynatrace.OpenTelemetry.Exporter
+namespace Dynatrace.OpenTelemetry.Exporter.Metrics
 {
     /// <summary>
     /// https://www.dynatrace.com/support/help/how-to-use-dynatrace/metrics/metric-ingestion/metric-ingestion-protocol/
     /// https://www.dynatrace.com/support/help/dynatrace-api/environment-api/metric-v2/post-ingest-metrics
     /// </summary>
     public class DynatraceMetricsExporter : MetricExporter
     {
-        internal readonly DynatraceExporterOptions Options;
-        private readonly ILogger<DynatraceMetricsExporter> logger;
-        private HttpClient httpClient;
-        private DynatraceMetricSerializer serializer;
+        private readonly DynatraceExporterOptions _options;
+        private readonly ILogger<DynatraceMetricsExporter> _logger;
+        private readonly HttpClient _httpClient;
+        private readonly DynatraceMetricSerializer _serializer;
 
         public DynatraceMetricsExporter(DynatraceExporterOptions options = null, ILogger<DynatraceMetricsExporter> logger = null)
         {
-            this.Options = options ?? new DynatraceExporterOptions();
-            this.logger = logger ?? NullLogger<DynatraceMetricsExporter>.Instance;
+            this._options = options ?? new DynatraceExporterOptions();
+            this._logger = logger ?? NullLogger<DynatraceMetricsExporter>.Instance;
             logger.LogDebug("Dynatrace Metrics Url: {Url}", options.Url);
-            this.httpClient = new HttpClient();
+            this._httpClient = new HttpClient();
             if (!string.IsNullOrEmpty(options.ApiToken))
             {
-                this.httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Api-Token", this.Options.ApiToken);
+                this._httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Api-Token", this._options.ApiToken);
             }
             var defaultLabels = new List<KeyValuePair<string, string>>();
-            if (options.Tags != null) defaultLabels.AddRange(options.Tags);
-            this.serializer = new DynatraceMetricSerializer(options.Prefix, defaultLabels);
+            if (options.DefaultDimensions != null) defaultLabels.AddRange(options.DefaultDimensions);
+            if (options.OneAgentMetadataEnrichment)
+            {
+                var enricher = new OneAgentMetadataEnricher(this._logger);
+                enricher.EnrichWithDynatraceMetadata(defaultLabels);
+            }
+            this._serializer = new DynatraceMetricSerializer(options.Prefix, defaultLabels);
```

Comment:
This seems like it has the same fallback order issue we talked about last week. A metric could have a label that overwrites the oneagent dimension.

Reply:
@pirgeo ping

Reply:
At the moment there is no de-duplication. Also, OneAgent dimensions are added after the default dimensions, in which case the OneAgent dimensions would overwrite the default dimensions, if there was deduplication. IIRC, this is the desired behavior.

Reply:
The next PR will also introduce the source as a static dimension, so you'll have to touch this again anyway.